src/asynqp/__init__.py

```diff
@@ -48,7 +48,13 @@ def connect(host='localhost',
         kwargs['port'] = port
 
     dispatcher = Dispatcher()
-    transport, protocol = yield from loop.create_connection(lambda: AMQP(dispatcher, loop), **kwargs)
+    try:
+        transport, protocol = yield from loop.create_connection(lambda: AMQP(dispatcher, loop), **kwargs)
+    except (ConnectionRefusedError, OSError):
+        # Throw a single exception instead of two
+        raise ConnectionRefusedError('Failed to connect - host: {} port: {}'
+                                     ' username: {} password: {} virtual_host: {}'
+                                     .format(host, port, username, password, virtual_host))
```

Comment:
Yes, it depends. _For that use case_ of simply retrying a connection for any type of failure, I agree that it's more convenient to catch one exception than two. However some clients may want to behave differently depending on the type of error.

For example, if the server actively refuses a connection it may be because Rabbit isn't running; whereas if the socket can't be opened it more likely indicates a local failure. By shoving both errors into one exception, we're ruling out the possibility of clients distinguishing the two types of failure; plus it's hard to go back to two exceptions because of backwards compatibility.

The `from e` trick does partially address this loss of information, but it's more designed for debugging than error handling.

If a client really does want to retry on _all_ failures, you can just catch any type of exception (`except Exception as e`).